@here/harp-mapview/lib/MapView.ts

```diff
@@ -2713,6 +2722,36 @@ export class MapView extends THREE.EventDispatcher {
             this.enqueueTextRendererCreation();
         }
 
+        // Check if this is the time to place the labels for the first time. Pretty much everything
+        // should have been loaded, and no animation should be running.
+        if (
+            !this.m_initialTextPlacementDone &&
+            !this.m_firstFrameComplete &&
+            !this.isDynamicFrame &&
+            !this.m_initialTextPlacementDone &&
+            !this.m_themeIsLoading &&
+            this.m_visibleTiles.allVisibleTilesLoaded &&
+            this.m_connectedDataSources.size + this.m_failedDataSources.size ===
+                this.m_tileDataSources.length &&
+            this.m_poiTableManager.finishedLoading
+        ) {
+            const textElementRendererFinished =
+                this.m_textElementsRenderer !== undefined &&
+                this.m_textElementsRenderer.ready &&
+                !this.m_textElementsRenderer.loading;
+
+            if (!this.textElementsRendererRequested || textElementRendererFinished) {
+                if (textElementRendererFinished) {
+                    // Placement of text shall be done now.
+                    this.m_textElementsRenderer!.placeAllTileLabels();
+                }
+                this.m_initialTextPlacementDone = true;
+            }
+            // Either render the labels after placing them, or wait for the TextElementsRenderer to
+            // be created (delayed creation).
+            this.update();
```

Comment:
This doesn't look right, that we've just almost said "frame complete"  and in very same time request new update.
So, is it compete or not ?

By _almost_ i mean If we're here, then following new "if" condition is always true
 * `this.m_initialTextPlacementDone` true, as it's set here
 * `!this.m_firstFrameComplete` -> true
 * `(this.m_textElementsRenderer === undefined ||
                (this.m_textElementsRenderer.ready && !this.m_textElementsRenderer.loading)) ` -> true as checked above 
 * `!this.m_animatedExtrusionHandler.isAnimating` -> may be always true if dynamic extrusion is disabled -> true

Reply:
Not calling update may let us without an update if the TER is finally created, of if the TER is finally loaded. As far as I can see, these conditions make sense.